### ISSUE TITLE
IPv6 support

### ISSUE THREAD

**Issue by JustAnotherArchivist** [ArchiveTeam/ArchiveBot#315]
ArchiveBot currently forces wpull to use IPv4 only:

https://github.com/ArchiveTeam/ArchiveBot/blob/33d2eb3112af19d32428c901d69a91e766d1b4c4/pipeline/archivebot/seesaw/wpull.py#L32

That's probably fine for now; IPv6-only websites are still rare. However, they will only become more popular with time, meaning we need to support IPv6 eventually. So I still need to ask: why is IPv6 disabled currently? What is necessary to enable IPv6 support besides removing that wpull option (assuming that the pipeline machines have IPv6 support)?

**Comment by JustAnotherArchivist** [comment]
Since we were having some issues with IPv6 again today, I dug around a bit. `--inet4-only` was originally added in commit 31d18948 (July 2014) as a workaround for chfoo/wpull#154. That issue has been fixed in wpull since April 2015 and version 1.2. So the original reason for that flag is no longer valid.

In the logs of #archivebot, I also found a mention that youtube-dl needs to be forced to IPv4 to avoid bans (from YouTube, I assume) on OVH and elsewhere (ivan on 2015-05-11). This is currently the case because wpull passes the corresponding `--force-ipv4` flag to youtube-dl when `--inet4-only` is specified. To preserve this while supporting archival of IPv6-only websites (i.e. removing wpull's `--inet4-only`), we'd have to add an option to wpull which controls the IPv4/IPv6 behaviour of youtube-dl independently of that of wpull as a whole. Unless there's another way to restrict youtube-dl to IPv4 that doesn't involve changing wpull (e.g. an environment variable?). However, I think we should evaluate whether that is indeed still necessary; I wouldn't be surprised if YouTube changed their policy in the meantime, for example.
